lib/usbd/usbd_transfer.c

```diff
@@ -720,7 +720,7 @@ static void _control_data_callback(usbd_device *dev,
 		return;
 	}
 
-	usbd_control_transfer_callback callback = transfer->user_data;
+	usbd_control_transfer_callback callback = (usbd_control_transfer_callback)transfer->user_data;
```

Comment:
But with some stricts - you can assign any pointer TO void* , 

but if you want assign some*  to typed* - need casts to destination type